Add type hints and data validation to the research graph script

genga6 opened the pull request:
researchgraph/evaluatenode/llmevalueate_node.py
researchgraph/retrievenode/github.py
・add pydantic.DirectoryPath and subprocess.CompletePrecess as type hints

fuyu-quant commented:
@genga6 
It would be helpful if you could set the following data validation.
- API input/output
- Data added to state by each Node

fuyu-quant commented on researchgraph/llmnode/llmnode.py:
```diff
@@ -17,7 +18,7 @@
 
 
-class State(TypedDict):
+class State(BaseModel):
```
I think it's fine to keep using TypedDict for LangGraph's State!
It looks like TypedDict is used in [the official documentation](https://langchain-ai.github.io/langgraph/tutorials/introduction/#part-3-adding-memory-to-the-chatbot), rather than pydantic.

fuyu-quant replied:
The same applies to other parts.

fuyu-quant commented:
@genga6 
openalex.pyの105行目にあるpyalexなどを実行した結果に対して、pydanticを使ったバリデーションを追加してほしい。
後述するように、”results”に何が含まれているのかが直感的にわかりづらいため、バリデーションを行うことで、どのような返値が返ってくるのかを確認でき、データ形式を直感的に理解できるようになります。
```
for search_term in keywords_list:
    results = works.search(search_term).get(
        page=1, per_page=self.num_retrieve_paper
    )
    all_search_results.append(results)
```

The image of the implementation is as follows.
```
class OpenAlexResponse(BaseModel):
　　paper_abstract: str
　　author: str
　　public_date:  datetime
```

genga6 commented:
**Summary**

This pull request adds type annotations throughout the codebase to enhance readability and type safety. Additionally, it introduces Pydantic data validation for API input and output to ensure data integrity and consistency.

**Changes Made**

Added type annotations to functions and methods.

Introduced Pydantic models for validating API input and output.

Enhanced error handling with Pydantic's validation errors to ensure robustness.

**Why This Change Matters**

Type Annotations: Helps developers understand function signatures and catch type-related issues during development.

Pydantic Data Validation: Provides clear and reliable data validation for API interactions, reducing the risk of runtime errors and unexpected data formats.

**Testing**

Manually tested API endpoints to ensure they correctly enforce data types and validation constraints.

Updated existing tests where needed to reflect type and validation changes.

Please let me know if you have any questions or need further clarifications!

fuyu-quant commented:
@genga6 
Please only change the parts I have added comments to.
Otherwise, LGTM
